src/DotNetty.Transport/Channels/ChannelDuplexHandler.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Transport.Channels
+{
+    /// <summary>
+    /// It is a good starting point if your <see cref="IChannelHandler"/> implementation needs to intercept operations and also
+    /// state updates.
+    /// </summary>
+    public class ChannelDuplexHandler : ChannelHandlerAdapter
+    {
+        /// <summary>
```

Comment:
I removed all code from this class. This class really needed? May be removed it, and change inheritance to ChannelHandlerAdapter just?